src/snowflake/snowpark/modin/plugin/_internal/index.py

```diff
@@ -103,6 +106,41 @@ def __init__(
                 tupleize_cols=tupleize_cols,
             )
 
+    def to_pandas_warning(self, func_name: str) -> None:
+        """
+        Helper method to notify users if they are using a method that currently calls to_pandas()
+        """
+        WarningMessage.single_warning(
+            f"The method {func_name} currently calls to_pandas() and materializes data. In future updates, this method will be lazily evaluated"
+        )
+
+    def is_lazy_check(func: Any) -> Any:
+        def check_lazy(*args: Any, **kwargs: Any) -> Any:
+            func_name = func.__name__
+            if args[0].is_lazy:
+                returned_value = func(*args, **kwargs)
+                args[0].to_pandas_warning(func_name)
```

Comment:
We should not warn here and keep the original warning in each method. Once we implement one method, we can remove the warning there. If you put warning here, how do we remove it one by one?

Reply:
ok makes sense, I will change it